src/ray/raylet/worker_pool.cc

```diff
@@ -1063,126 +1061,73 @@ void WorkerPool::TryKillingIdleWorkers() {
       RAY_LOG(DEBUG) << "idle worker is already dead. Not going to kill worker "
                      << idle_worker->WorkerId();
       // This worker has already been killed.
-      // This is possible because a Java worker process may hold multiple workers.
```

Comment:
nit: improve comment to say when this is possible? suggestion below which may not be correct
> This is possible if the idle_worker has been killed but hasn't been removed from the idle list yet.